### PR TITLE
Ensures that we always set baggage in scope when setting a value

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
fixes gh-181

### CODE REVIEW

File: micrometer-tracing/src/main/java/io/micrometer/tracing/Baggage.java [hunk 1]
```diff
@@ -86,4 +100,23 @@ public BaggageInScope makeCurrent() {
      */
     BaggageInScope makeCurrent();
 
+    /**
+     * Sets the current baggage in scope with given value.
+     * @param value to set
+     * @return a {@link BaggageInScope} instance
+     */
+    default BaggageInScope makeCurrent(String value) {
```
Comment (jonatan-ivanov) [inline micrometer-tracing/src/main/java/io/micrometer/tracing/Baggage.java:108]: I'm debating with myself if the extra behavior (setting the value) should be called out in the method name or not. I like the overloads though.

File: micrometer-tracing-tests/micrometer-tracing-test/src/main/java/io/micrometer/tracing/test/simple/SimpleBaggageManager.java [hunk 2]
```diff
@@ -104,10 +106,21 @@ private Baggage createSimpleBaggage(String name) {
     }
 
     @Override
+    @Deprecated
```
Comment (jonatan-ivanov) [inline micrometer-tracing-tests/micrometer-tracing-test/src/main/java/io/micrometer/tracing/test/simple/SimpleBaggageManager.java:109]: Could you please add a note to use `createBaggageInScope` instead?

File: micrometer-tracing-tests/micrometer-tracing-integration-test/src/main/java/io/micrometer/tracing/test/reporter/wavefront/WavefrontBraveSetup.java [hunk 3]
```diff
@@ -54,6 +39,15 @@
 import io.micrometer.tracing.reporter.wavefront.WavefrontSpanHandler;
 import io.micrometer.tracing.test.reporter.BuildingBlocks;
 
+import java.util.Deque;
```
Comment (jonatan-ivanov) [inline micrometer-tracing-tests/micrometer-tracing-integration-test/src/main/java/io/micrometer/tracing/test/reporter/wavefront/WavefrontBraveSetup.java:42]: Which import order is the correct?

File: micrometer-tracing-tests/micrometer-tracing-test/src/main/java/io/micrometer/tracing/test/simple/SimpleBaggageManager.java (hunk 2 above)
Comment (marcingrzejszczak) [reply micrometer-tracing-tests/micrometer-tracing-test/src/main/java/io/micrometer/tracing/test/simple/SimpleBaggageManager.java:109]: But this are inheriting from a method that already provides that info

File: micrometer-tracing-tests/micrometer-tracing-integration-test/src/main/java/io/micrometer/tracing/test/reporter/wavefront/WavefrontBraveSetup.java (hunk 3 above)
Comment (marcingrzejszczak) [reply micrometer-tracing-tests/micrometer-tracing-integration-test/src/main/java/io/micrometer/tracing/test/reporter/wavefront/WavefrontBraveSetup.java:42]: I have no idea. I haven't noticed that I have changed that class

**Comment by marcingrzejszczak** [comment]
Done via https://github.com/micrometer-metrics/tracing/commit/7e88a7aa56f6e73a048b007ecea6a91307b45590

**Pull request closed without merging**